app/views.py

```diff
@@ -241,5 +260,71 @@ def checkvalid(machine, card):
     else:
         return("no result!")
     '''
+'''
+LOGGING
+'''
+
+@app.route('/log/new', methods=['POST'])
+def logusage():
+    try:
+        # Force parsing as JSON even if content header is
+        # incorrect, because we don't accept anything else.
+        json = request.get_json(force=True)
+    except Exception as ex:
+        # Return a bad request if we aren't getting our JSON.
+        # This could happen if the content type isn't set properly or if the JSON
+        # is particularly malformed such that it can't even be
+        print ("Bad JSON request.")
+        print(ex)
+        return "{\"error\":\"Bad JSON\"}", 400 # 400 BAD REQUEST
+
+    try:
+        log = Log()
+        log.endtime = datetime.datetime.now()
+        log.starttime = log.endtime - datetime.timedelta(seconds=int(json["elapsed"]))
```

Comment:
Might be a good idea to validate that `elapsed` is a positive number, and if it's not, reject the request and log it in such a way that it's easy to see that somebody was submitting dodgy data